diff --git a/includes/abstracts/abstract-ur-field-settings.php b/includes/abstracts/abstract-ur-field-settings.php
index 3f02c310..ead8a14b 100644
--- a/includes/abstracts/abstract-ur-field-settings.php
+++ b/includes/abstracts/abstract-ur-field-settings.php
@@ -53,8 +53,9 @@ abstract class UR_Field_Settings {
 
 		if ( isset( $this->field_data->advance_setting->$key ) ) {
 			return $this->field_data->advance_setting->$key;
+		} elseif( isset( $this->field_data->general_setting->$key ) ) {
+			return $this->field_data->general_setting->$key;
 		}
-
 		return '';
 	}
 
diff --git a/includes/abstracts/abstract-ur-form-field.php b/includes/abstracts/abstract-ur-form-field.php
index c5d0d7a2..84cc0256 100644
--- a/includes/abstracts/abstract-ur-form-field.php
+++ b/includes/abstracts/abstract-ur-form-field.php
@@ -84,12 +84,8 @@ abstract class UR_Form_Field {
 		if ( isset( $this->admin_data->advance_setting->$key ) ) {
 			return $this->admin_data->advance_setting->$key;
 		}
-
-		if ( isset( $this->field_defaults[ 'default_' . $key ] ) ) {
-			return $this->field_defaults[ 'default_' . $key ];
-		}
-
-		return '';
+		//fallback to general setting data if advance setting not found.
+		return $this->get_general_setting_data( $key );
 	}
 
 	/**
diff --git a/includes/functions-ur-core.php b/includes/functions-ur-core.php
index 25d7a4a5..92c1dc42 100644
--- a/includes/functions-ur-core.php
+++ b/includes/functions-ur-core.php
@@ -907,15 +907,6 @@ function ur_get_general_settings( $id ) {
 			'required'    => true,
 			'tip'         => __( 'Enter text for the form field description.', 'user-registration' ),
 		),
-		'field_name'  => array(
-			'setting_id'  => 'field-name',
-			'type'        => 'text',
-			'label'       => __( 'Field Name', 'user-registration' ),
-			'name'        => 'ur_general_setting[field_name]',
-			'placeholder' => __( 'Field Name', 'user-registration' ),
-			'required'    => true,
-			'tip'         => __( 'Unique key for the field.', 'user-registration' ),
-		),
 
 		'placeholder' => array(
 			'setting_id'  => 'placeholder',
